open_academy/models/session.py

```diff
@@ -21,19 +24,25 @@ def _compute_taken_seats(self):
                 taken_seats = round((len(record.attendees_ids) * 100) / record.number_of_seats)
             record.taken_seats = taken_seats
 
-    @api.onchange('number_of_seats','attendees_ids')
+    @api.onchange('number_of_seats', 'attendees_ids')
     def _onchange_invalid_values(self):
         if self.number_of_seats < 0:
             return {
                 'warning': {
-                'title': "Something bad happened.",
-                'message': "Number of seats must be greater than 0."
-                } 
+                    'title': "Something bad happened.",
+                    'message': "Number of seats must be greater than 0."
+                }
             }
         if len(self.attendees_ids) > self.number_of_seats:
             return {
                 'warning': {
-                'title': "Something bad happened.",
-                'message': "Number of attendees cannot be greater than the number of available seats."
-                } 
+                    'title': "Something bad happened.",
+                    'message': "Number of attendees cannot be greater than the number of available seats."
+                }
             }
+
+    @api.constrains('attendees_ids')
+    def _check_attendees_ids(self):
+        for key, value in enumerate(self.attendees_ids):
+            if self.instructor_id.id == self.attendees_ids[key].id:
+                raise ValidationError("An instructor cannot be an attendee of his own session")
```

Comment:
Could you try something like this?

```suggestion
        if self.instructor_id.id in self.atendees_ids:
                raise ValidationError("An instructor cannot be an attendee of his own session")
```